sips/fork_support.md

```diff
@@ -20,56 +20,231 @@ Client implementation should calculate when a new duty starts to decide if it st
+	ShifuV2Testnet = DomainType{0x0, 0x0, byte(ShifuNetworkID), 0x1}
+	JatoTestnet    = DomainType{0x0, 0x0, byte(JatoNetworkID), 0x1}
+	JatoV2Testnet  = DomainType{0x0, 0x0, byte(JatoV2NetworkID), 0x1}
 )
 
+// ForkData is a simple structure holding fork information for a specific chain (and its fork)
 type ForkData struct {
-    Epoch  phase0.Epoch
-    Domain DomainType
+	// Epoch in which the fork happened
+	Epoch phase0.Epoch
+	// Domain for the new fork
+	Domain DomainType
+}
+
+func (domainType DomainType) GetNetworkID() NetworkID {
+	return NetworkID(domainType[2])
 }
 
-func (domainType DomainType) GetChain() byte {
-    return domainType[2]
+func (networkID NetworkID) GetForksData() []*ForkData {
+	switch networkID {
+	case MainnetNetworkID:
+		return mainnetForks()
+	case JatoNetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoTestnet}}
+	case JatoV2NetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoV2Testnet}}
+	default:
+		return []*ForkData{}
+	}
 }
 
-func (domainType DomainType) GetForksData() []*ForkData {
-    switch domainType.GetChain() {
-    case GenesisChain:
-        return genesisForks()
-    default:
-        return []*ForkData{}
-    }
+// mainnetForks returns all forks for the mainnet chain
+func mainnetForks() []*ForkData {
+	return []*ForkData{
+		{
+			Epoch:  0,
+			Domain: GenesisMainnet,
+		},
+	}
 }
 
-func genesisForks() []*ForkData {
-    return []*ForkData{
-        {
-            Epoch:  0,
-            Domain: GenesisMainnet,
-        },
-    }
+func (networkID NetworkID) DefaultFork() *ForkData {
+	return networkID.GetForksData()[0]
 }
 
+// GetCurrentFork returns the ForkData with highest Epoch smaller or equal to "epoch"
+func (networkID NetworkID) ForkAtEpoch(epoch phase0.Epoch) (*ForkData, error) {
+	// Get list of forks
+	forks := networkID.GetForksData()
+
+	// If empty, raise error
+	if len(forks) == 0 {
+		return nil, errors.New("Fork list by GetForksData is empty. Unknown Network")
+	}
+
+	var current_fork *ForkData
+	for _, fork := range forks {
+		if fork.Epoch <= epoch {
+			current_fork = fork
+		}
+	}
+	return current_fork, nil
+}
+
+func (f ForkData) GetRoot() ([]byte, error) {
+	byts, err := json.Marshal(f)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal ForkData")
+	}
+	ret := sha256.Sum256(byts)
+	return ret[:], nil
+}
 ```
 
+### Structures access to types
+
+- `Share`: It's more appropriate for the structure to hold the information on the network that it's in, instead of the domain type.
+    Thus, _Share_ will contain a `NetworkID` field instead of a `DomainType` field.
+
+Regarding the indirect usage of Identifiers:
+- `Instance`:
+  - Must keep its identifier to send properly formed messages.
+- `Controller`:
+  - Should have its _Identifier_ and _DomainType_ updated when a new fork occurs. For that, it will have setter functions for both fields. These functions will be used by the `Runner` to update the controller.
+
+```go
+
+type Controller struct {
```

Comment:
please update to current controller struct.. this seems old

Reply:
Indeed. Updating, thanks